### PR TITLE
Version 5

### LINKED ISSUE #48: Customizable Keybindings
For most vim plugins, I don't use default bindings. A big reason is because I use colemak and don't have jk (n and e are in the j and k positions).

For example, I do this for unite:
```VimL
autocmd FileType unite call s:unite_settings()
function! s:unite_settings()
	nmap <buffer> <ESC> <Plug>(unite_exit)
	" nmap <buffer> n j
	nmap <buffer> n <Plug>(unite_skip_cursor_down)
	" nmap <buffer> e k
	nmap <buffer> e <Plug>(unite_skip_cursor_up)
endfunction
```
I realize that this doesn't work for several reasons with ctrlspace (the ctrlspace buffer does not seem to have a filetype, there are no \<Plug\>s, and something like "nmap <buffer> n j" won't work in the first place). Is there a recommended way to change the default bindings? 

For people who don't use QWERTY and for those who like to choose their mappings a way to do this without having to manually go through the plugin and change them all would be convenient.

### LINKED ISSUE #96: add 'Vim tabs madness' article to documentation
As I didn't find 'Buffers, Windows, and Tabs' section helpful (have started using vim a few weeks ago), I was looking for more down-to-earth explanation of buffers/windows/tabs concepts and this article turned out to be very helpful:

https://joshldavis.com/2014/04/05/vim-tab-madness-buffers-vs-tabs/

### LINKED ISSUE #97: add screenshots of workspace(s) from a real project
I'm sure that the illustration displaying the relationships is very good when you're already familar with the plugin, but from my perspective (vim newbie, using RubyMine) is not enough. Some screenshots / images would add more value.

### LINKED ISSUE #102: Using fzf for fuzzy finding
Hi,

It seems to me the fuzzy search is a little slow in large project. In fact, there is a super fast fuzzy finding (filtering) engine: [fzf](https://github.com/junegunn/fzf). I'm wondering if you can migrate the fuzzy finding to using fzf, or if you could provide an example how I can configure ctrlspace to use fzf?
Thanks.

### LINKED ISSUE #108: Command to finish task directly from search?
In `search` mode it seems that `<CR>` merely brings us back to the parent mode upon which another `<CR>` will have the effect that was desired at search time as it were.

screen shot 2015-03-08 at 9 37 31 pm

It would be nice to to support a new command like shift-enter that can just complete our action right then and there.

### LINKED ISSUE #112: Opened wrong files in workspace relative to project "root"
I have a folder structure:

    root/
        .git/
        project1/
        project2/

When I work in `project1`, my `:pwd` is `project1`. I'm also using [vim-project](https://github.com/amiorin/vim-project) which sets the pwd to `project1/` if you're editing a file in a subfolder of `project1` (very nice).

I saved a workspace in one vim session, with lots of files open like `root/project1/folder/file.txt`, with Vim showing me the relative path as `folder/file.txt`which is correct.

Then I opened Vim, switched to my project with vim-project (which basically sets `:pwd` to `project1/`, then I loaded the workspace. Now all of my paths are wrong because it mangled the relative paths. It tried to load relative path `project1/folder/file.txt` when it should be loading `folder/file.txt` which is correct relative to the `:pwd`.

I'm not sure if the workspace is saved relative to a git directory which is why I included the git folder in the example structure. I have multiple "projects" as a subdirectory of one git repo, and ctrlspace doesn't seem to handle that.

### LINKED ISSUE #122: File content deleted after move or rename
After renaming a file in the buffer list, the file content gets deleted. This happens if the renamed file is not the current active buffer.

### PR DESCRIPTION
**Pull request by szw**
### Vim-CtrlSpace 5.0.0 
Finally... After 5 months I'm happy to announce Version 5. Everyone is welcome to test this candidate. In case of Vundle it's as easy as:
```
cd ~/.vim/bundle/vim-ctrlspace
git fetch
git checkout version_5
```
Then in Vim:
```
:PluginUpdate vim-ctrlspace
```

Don't forget to restart Vim afterwards. Additionally update your `.vimrc` to match new configuration syntax. For more details please check `:help ctrlspace-version-5`.

This release:

* Fixes #108 
* Fixes #122 
* Fixes #102 
* Fixes #97 
* Fixes #96
* Fixes #112
* Fixes #48

Version 5 is not backward compatible but workspaces saved in Version 4.x should work. However, workspaces saved in Version 5 won't be loaded correctly in Version 4.x.

@ryrych Proof reading of docs would be really appreciated :) Thanks!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,1633 +1,287 @@
...
-| `⌕`     | `*`    | Buffer      | Zoom Mode indicator     |
-| `○●○`   | `-+-`  | Tab         | Tab List indicator      |
-| `|∷|`   | `|::|` | Workspace   | Workspace Load Mode     |
-| `[∷]`   | `[::]` | Workspace   | Workspace Save Mode     |
-| `♥`     | `BM`   | Bookmark    | Bookmark List indicator |
+##### Upgrading from Version 4
 
-Items listed in the plugin window can have additional indicators (following the
-item text):
+Version 5 is not backward compatible. But main issues are
```
Comment (ryrych) [inline README.md]: Most issues are related…

File: README.md [hunk 2]
```diff
@@ -1,1633 +1,287 @@
...
+related to restoring your workspaces and there is a simple workaround for it.
 
-| Unicode | ASCII | Indicator                         |
-|:-------:|:-----:| --------------------------------- |
-| `+`     | `+`   | Item modified                     |
-| `★`     | `*`   | Item active                       |
-| `☆`     | `-`   | Item visible or previously active |
+After upgrading please open `cs_workspaces` file for given project (it's
```
Comment (ryrych) [inline README.md]: for a given project

File: README.md [hunk 3]
```diff
@@ -1,1633 +1,287 @@
...
-Enables you to provide your own symbols. It's useful if for example your font
-doesn't contain enough symbols or the glyphs are poorly rendered. 
-
-Default value:
-
-```VimL
-if g:ctrlspace_unicode_font
-  let g:ctrlspace_symbols = {
-        \ "cs":      "⌗",
-
```
Comment (ryrych) [inline README.md]: Vim-Ctrlspace displays icons in the UI if your font supports UTF-8, or ASCII characters as a fallback. Some symbols (glyphs) might not look well with the font you are using, so feel free to change and adjust them.

This is the config I use for *Inconsolata* font in 

[your setup]

If you feel that you have found a better symbol for a given view, you are more than welcome to open a pull request.

Comment (ryrych) [inline README.md]: tweak up => adjust / change

**Comment by rgrinberg** [comment]
Does ctrlspace still depend on ruby bindings? I'd like to use ctrlspace with neovim but it doesn't support ruby.

**Comment by szw** [comment]
Nope, this new version does not depend on Ruby. Actually it doesn't require any bindings at all. All communication with optional Go engine is done by pure `system()` calls.

**Comment by rgrinberg** [comment]
:100:

**Comment by lazywei** [comment]
awesome!

**Comment by milin** [comment]
Looks like a bunch of text in the README got removed. Does the old README not apply anymore? It had some helpful information about how to create tabs and grouping buffers together.

**Comment by szw** [comment]
Check the detailed Vim help file. I'm sure a lot of information landed there: `:help ctrlspace`

**Comment by milin** [comment]
Thanks, all the info in there in the Vim help file.

**Pull request merged**